plugins/analytics/public/circl_passive_ssl.py

```diff
@@ -0,0 +1,132 @@
+                             observable.value, auth=auth,
+                             headers={
+                                 'User-Agent': 'Yeti Analytics Worker',
+                                 'accept': 'application/json'
+                             },
+                             proxies=yeti_config.proxy)
+
+            if r.ok:
+                return r.json()
+
+    @staticmethod
+    def fetch_cert(cert_sha1, settings):
+        auth = (
+            settings['circl_username'],
+            settings['circl_password']
+        )
+
+        r = requests.get(CirclPassiveSSLApi.API + 'cfetch/' + cert_sha1,
+                         auth=auth,
+                         headers={
+                             'User-Agent': 'Yeti Analytics Worker',
+                             'accept': 'application/json'
+                         },
+                         proxies=yeti_config.proxy)
+
+        if r.ok:
+            return r.json()
+
+
+class CirclPassiveSSLSearchIP(OneShotAnalytics, CirclPassiveSSLApi):
+    default_values = {
+        'name': 'Circl.lu IP to ssl certificate lookup.',
+        'group': 'SSL Tools',
+        'description': 'Perform a lookup on ssl certificates \
+         related to an ip address.'
+    }
+
+    ACTS_ON = ['Ip']
+
+    @staticmethod
+    def analyze(observable, results):
+        links = set()
+
+        if isinstance(observable, Ip):
+            ip_search = CirclPassiveSSLApi.search_ip(
+                observable, results.settings)
+            if ip_search:
+                json_string = json.dumps(
+                    ip_search, sort_keys=True, indent=4,
+                    separators=(',', ': '))
+
+                results.update(raw=json_string)
+
+                for ip_addr, ip_details in ip_search.items():
+                    for cert_sha1 in ip_details.get('certificates', []):
+                        try:
+                            cert_result = CirclPassiveSSLApi.fetch_cert(
+                                cert_sha1, results.settings)
+                            if cert_result:
+                                _info = cert_result.get('info', {})
+                                x509 = load_certificate(
+                                    FILETYPE_PEM, cert_result.get('pem'))
+
+                                DER = dump_certificate(FILETYPE_ASN1, x509)
+
+                                cert_ob = Certificate.from_data(
+                                    data=DER)
+
+                                subject = CertificateSubject.get_or_create(
+                                    value=_info.get('subject', ''))
+
+                                issuer = CertificateSubject.get_or_create(
+                                    value=_info.get('issuer', ''))
+
+                                links.update(observable.active_link_to(
+                                    cert_ob,
+                                    'cert_details',
```

Comment:
I'd suggest the use of the same names that are already used in IVRE plugin:
```suggestion
                                    'ssl-cert',
```